proposals/bip-001.md

```diff
@@ -0,0 +1,107 @@
+# BIP Purpose and Guidelines
+
+| Field | Value |
+| --- | --- |
+| BIP | 1 |
+| Title | BIP Purpose and Guidelines |
+| Author | Alfredo Beaumont |
+| Status | Draft |
+| Created | 2017-05-26 |
+| Updated | 2017-05-26 |
+
+## Abstract
+
+BIP stands for Babelfish Improvement Proposal. A BIP is a design document which
+describes a change proposal to Babelfish. It's initially designed specifically
+to propose changes to [UAST](https://doc.bblf.sh/uast/specification.html)
+components regarding code categorization through Roles or related entities, but
+its scope may become broader in the future. It's informally defined.
```

Comment:
I think that you're right about the utils but changed in the driver creation workflow, SDK public API (even if they don't break backwards compatibility, they could polute the API) should be included (it's Babelfish IP after all). We could probably add a "Scope" section indicating what parts of babelfish the BIP touches.